### ISSUE TITLE
Dead node on 11 node 32 cpu cluster 

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#31738]
**Describe the problem**

Dead node on 11 node 32 cpu cluster

![image](https://user-images.githubusercontent.com/22278911/47367292-67905a80-d6ad-11e8-8daf-9fea1d4a744d.png)

**To Reproduce**

```diff
@@ -403,7 +403,7 @@ func loadTPCCBench(
        }
 
        partArgs := ""
-       rebalanceWait := time.Duration(b.LoadWarehouses/100) * time.Minute
+       rebalanceWait := time.Duration(b.LoadWarehouses/25) * time.Minute
        switch b.LoadConfig {
        case singleLoadgen:
                t.l.Printf("splitting and scattering\n")
@@ -677,11 +677,11 @@ func registerTPCCBench(r *registry) {
                        // StoreDirVersion: "2.0-5",
                },
                {
-                       Nodes: 3,
-                       CPUs:  16,
+                       Nodes: 11,
+                       CPUs:  32,
 
-                       LoadWarehouses: 2000,
-                       EstimatedMax:   1300,
+                       LoadWarehouses: 10000,
+                       EstimatedMax:   9000,
```

`bin/roachtest bench '^tpccbench/nodes=11/cpu=32$$' --wipe=false --user=andy`

**Expected behavior**
No node crashing

**Additional data / screenshots**

**Environment:**
 - CockroachDB version 2.1 rc 1

`andy-1540300582-tpccbench-nodes-11-cpu-32`

[11 nodes.zip](https://github.com/cockroachdb/cockroach/files/2506513/11.nodes.zip)

**Comment by tschottdorf** [comment]
@awoods187 let's close this for #31409, you'll likely see this same kind of crash on any configuration. I'm going to need to concentrate on fixing #31409.

**Comment by awoods187** [comment]
But this didn't fail when I ran it Friday on 10/15 beta (without the ramp time increase) and its not with partitioning. What makes you think its the same as #31409?

**Comment by tschottdorf** [comment]
Wow, a double whammy! First of all, I see memory growth that seems similar to #31409:

![image](https://user-images.githubusercontent.com/5076964/47377415-7f8bcc80-d6f5-11e8-8400-20bbdcc7f672.png)

The last runtime stats are

> I181023 13:59:54.707791 499 server/status/runtime.go:465  [n4] runtime stats: 22 GiB RSS, 596 goroutines, 9.0 GiB/4.1 GiB/14 GiB GO alloc/idle/total, 7.3 GiB/8.6 GiB CGO alloc/total, 83243.4 CGO/sec, 59.0/31.3 %(u/s)time, 0.0 %gc (1x), 285 MiB/230 MiB (r/w)net

But additionally there isn't a heap profile. Why not? Well, the node panicked. You need to run with https://github.com/cockroachdb/cockroach/pull/31734 which @nvanbenschoten just merged (my 2.1 cluster also hit this crash).

```
I181023 13:59:58.321234 103211 storage/replica.go:874  [n4,replicaGC,s4,r14135/3:/Table/58/1/1625/88541{-/0}] removed 9 (0+9) keys in 16ms [clear=15ms commit=1ms]
E181023 13:59:58.464624 185 util/log/crash_reporting.go:477  [n4,s4] Reported as error 8e861b5f166a4557a54af9eee406361d
panic: runtime error: index out of range [recovered]
        panic: runtime error: index out of range

goroutine 185 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420b003f0, 0x2ff6060, 0xc420ceaa80)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:184 +0x11f
panic(0x273b600, 0x41dbd80)
        /usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/storage.DecodeRaftCommand(0x0, 0x0, 0x0, 0x3, 0x2, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_raftstorage.go:1059 +0x19e
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeTrackForwardedProposalLocked(0xc4651e4e00, 0xc483f17180, 0xc62bdb5458)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:4049 +0x217
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).stepRaftGroup.func1(0xc483f17180, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3992 +0x197
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked.func1(0xc45da43b88, 0x4c65, 0xc483f17180, 0x6c1b00, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:576 +0x30
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc4651e4e00, 0x2bfc900, 0xc45da43b88, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:577 +0xc2
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroup(0xc4651e4e00, 0x0, 0xc45da43b88, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:602 +0x8c
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).stepRaftGroup(0xc4651e4e00, 0xc62bdb5458, 0x0, 0xc426af96c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3973 +0x55
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRaftRequestWithReplica(0xc420a40580, 0x2ff6060, 0xc5938b1ce0, 0xc4651e4e00, 0xc62bdb5458, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3554 +0xeb
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRequestQueue.func1(0x2ff6060, 0xc5938b1ce0, 0xc4651e4e00, 0x2ff6060)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3861 +0x97
github.com/cockroachdb/cockroach/pkg/storage.(*Store).withReplicaForRequest(0xc420a40580, 0x2ff6060, 0xc5938b1ce0, 0xc62bdb5458, 0xc45da43ed0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3517 +0x135
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRequestQueue(0xc420a40580, 0x2ff6060, 0xc420ceaa80, 0x4c65)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3855 +0x229
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42058aa00, 0x2ff6060, 0xc420ceaa80)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:225 +0x21b
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x2ff6060, 0xc420ceaa80)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x3e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420327de0, 0xc420b003f0, 0xc420327dd0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0xe9
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xad
```

**Comment by tschottdorf** [comment]
@awoods187 please reopen if this configuration fails again.

BTW, you can also find out about these proximate causes yourself: open the admin UI, find the dead node, ssh to it, look at the tail of the log. If there's a stack trace, that's what I'm going to want to see. If there isn't, run `sudo dmesg` and check if the last couple of lines talk about "sacrificing a child" (or just blindly paste them in the issue), in which case it's an OOM.
